### DISCUSSION TITLE
Option to change output directory

### DISCUSSION THREAD

**Discussion by hengsin** [JaneWardSandy/eclipse-pde-partial-idea#66] [Q&A] [answered]
Hi, is there any way to change the default output folder from out and out/production to target and target/classes? Thanks.

**Comment by JaneWardSandy** [comment]
Not currently supported, I will try to provide tuning options in the next version.

**Reply by JaneWardSandy** [reply]
#67

**Reply by hengsin** [reply]
Related to this, I'm not sure is my mistake or it doesn't seems to behave the same as the default compiler that output to target/classes. It seems non-Java elements (css, js, images, xml, etc) doesn't get copy from the source folder to output/production.

**Reply by JaneWardSandy** [reply]
It seems you are using `Maven` as build tool? I have not used `Maven` to develop PDE Plugins, so I haven't check the compatible  between this plugin and Maven or BND.

As `Eclipse Equinox Platform` logic, it will read class from output/production when it was written in `dev.properties`, static file will be read under module root directory.
And code compiler was using built-in compiler by IntelliJ IDEA, it was copy static file only in `src` directory.

<img width="383" alt="image" src="https://user-images.githubusercontent.com/15345355/154612089-4a3a4832-15f6-4300-8c43-99b20e728d4c.png">


So, if you use `Maven` or BND, `Gradle`, it maybe incompatible. 
And I have no plan to support `Maven`, BND, `Gradle` yet.

**Reply by hengsin** [reply]
Alright, will check again. I'm using dev.properties and my static file does reside in src directory (with package though). The out/production thing is problematic for project that uses maven (I'm using maven tycho) since the default output directory of maven is target and target/classes. If I make any changes to the pom and reload it, IntelliJ will change the output path from out/production to target/classes.

**Reply by JaneWardSandy** [reply]
OK, I will add config to change the default compile path, then you can check it.
After that, we can try to find or make it compatible.

**Comment by JaneWardSandy** [comment, marked as answer]
This feature has been fix and release at v1.3.6
